File: swift/Sources/prefix32.swift
//
//import Prefix from './prefix';
//import IpBits from './ip_bits';

class Prefix32 {
    class func from(_ my: Prefix, _ num: UInt8) -> Prefix? {
        return Prefix32.create(num);
    }

    class func create(_ num: UInt8) -> Prefix? {
        if (0 <= num && num <= 32) {
            let ip_bits = IpBits.v4();
            let bits = ip_bits.bits;
            return Prefix(
                num: num,
                ip_bits: ip_bits,
                net_mask: Prefix.new_netmask(num, bits),
                vt_from: Prefix32.from
            );
        }
        return nil;
    }

}


//export default Prefix32;

File: js/src/prefix32.ts
import Prefix from "./prefix.js";
import IpBits from "./ip_bits.js";

const Prefix32 = {
  from(my: Prefix, num: number): Prefix {
    return Prefix32.create(num);
  },

  create(num: number): Prefix {
    if (num >= 0 && num <= 32) {
      // static _FROM: &'static (Fn(&::prefix::Prefix, usize) -> Result<::prefix::Prefix, String>) =
      // &from;
      // static _TO_IP_STR: &'static (Fn(&Vec<u16>) -> String) = &to_ip_str;
      const ip_bits = IpBits.v4();
      const bits = ip_bits.bits;
      return new Prefix({
        num,
        ip_bits,
        net_mask: Prefix.new_netmask(num, bits),
        vt_from: Prefix32.from,
        // vt_to_ip_str: _TO_IP_STR,
      });
    }
    return null;
  },
};

export default Prefix32;
